### PR TITLE
#923 Remove ConsoleWriterFactory

### PR DESCRIPTION
**Pull request by pljones**
All instances of `ConsoleWriterFactory` have been removed.

In addition, on Windows (when not headless), the console has been attached once on start up to ensure messages appear.  This only happens when run from the command line - otherwise the messages don't appear.

Parameterised messages use `QString` formatting but that's needed an extra `qUtf8Printable` call to pass the result to `qInfo()` etc.

A few of the start up messages were warnings, so they're now `qWarning()`, rather than `qInfo()`.  Start up failure is now displayed as `qCritical()` messages.

As we've generally moved to `qInfo()` etc, I removed the QT version check where it was in place.

----
Unrelated bits...
* I changed the help text for `--centralserver` to use the term "server list".
* I fixed the formatting of `--ctrlmidich` and `--clientname` to be consistent with the other long opt-only options.

### CODE REVIEW

**Comment by pljones** [comment]
Tests:

- tested start up with every parameter (`-v` last) and all the messages formatted ok
- started a linux server and a Windows client and connected to it - all messages formatted ok
- sent a "start new recording" signal to the server - messages formatted ok
- sent a "toggle recording" signal to the server - messages formatted ok
- registered a server at a serverlist (found a bug...) - messages formatted ok (now)

Anything else anyone can think of?

**Comment by pljones** [comment]
@softins @ann0see - can you review?

**Comment by softins** [comment]
Will do, but it will either be later today or else tomorrow.

**Comment by ann0see** [comment]
I can look into it tomorrow

**Comment by pljones** [comment]
Thanks :)

File: src/main.cpp
Comment (ann0see) [inline src/main.cpp:90]: I assume this is the first time we change coding style?

**Review by ann0see — COMMENTED** [review]
It's quite a big change. It does compile, I think but should we test it before a merge again?

**Review by ann0see — APPROVED** [review]
Still to be tested (real world test), but since you couldn't find issues, I will approve it

**Comment by softins** [comment]
I saw in several places the comment `// TODO we should use the ConsoleWriterFactory() instead of qInfo()`, so I'm trying to understand the reasons for firstly that comment, and now the opposite decision. I see `ConsoleWriteFactory` says it was implemented by @pljones 

What environments has compilation been tested on so far? I can test on Linux(RPi) and Mac(Mojave), but not (yet) PC.

**Comment by ann0see** [comment]
The CodeQL check makes sure that it compiles on all 3 OS. So this should be ok. 

Raspberry Pi is untested but I‘m quite sure if it compiled on Debian and Ubuntu (both amd64) armhf should work too n

**Comment by softins** [comment]
> The CodeQL check makes sure that it compiles on all 3 OS. So this should be ok.

I have much to learn! Not sure what this is.

> Raspberry Pi is untested but I‘m quite sure if it compiled on Debian and Ubuntu (both amd64) armhf should work too.

Just trying this now.

**Review by softins — APPROVED** [review]
Compiles and runs fine on RPi too, so approving.

Would still like to understand the reason for originally moving to `ConsoleWriterFactory` and why this no longer applies.

**Comment by softins** [comment]
It would appear that my review doesn't count - presumably I don't have write access:
> At least 2 approving reviews are required by reviewers with write access.

**Comment by ann0see** [comment]
Does it work now? I just changed the role maindeveloper role to have write access and made you an organisation admin. 

I think we can merge this now

**Comment by softins** [comment]
> Does it work now? I just changed the role maindeveloper role to have write access and made you an organisation admin.

Thanks, github is now happy with my approval.

> I think we can merge this now

I agree.

**Comment by ann0see** [comment]
Merged.

**Comment by ann0see** [comment]
> I have much to learn! Not sure what this is.

It basically tests if the code compiles and to some extent checks it for vulnerabilities.

**Comment by pljones** [comment]
> Would still like to understand the reason for originally moving to `ConsoleWriterFactory` and why this no longer applies.

Qt4 compatibility.  We gave up on that some time last year, I think.

**Comment by pljones** [comment]
> 
> 
> Merged. @pljones I think we should still do some real world tests.

I did
```
./Jamulus -n -i foo.ini -p 55850 -t -d -e localhost -f '[1.2.3]' -l /dev/stdout -L -m /dev/stdout -o 'here;here;23' -R /tmp --norecord -s -T -u 1 -w 'Hello world' -z -M --mutemyown -c 1 -j --ctrlmidich 13 --clientname foo -l /dev/stdout
```
before the push and it seemed to work okay -- all the main.cpp messages appeared - also adding -v stopped start up still.

I also did
```
./Jamulus -n -p 55850 -d -e localhost -f '[3.0.0]' -R /tmp -s -T -u 1 -c 1
```
and tested the various recorder messages with SIGUSR1, SIGUSR2 and clients connecting.

**Comment by softins** [comment]
> > Would still like to understand the reason for originally moving to `ConsoleWriterFactory` and why this no longer applies.
> 
> Qt4 compatibility. We gave up on that some time last year, I think.

Ah, cool, understood. Thanks! I remember that happening.

**Pull request merged**
